Accessing the public share link to a private poll with the owner account breaks access

### ⚠️ This issue respects the following points: ⚠️

- [x] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [x] This issue is **not** already reported on Github _(I've searched it)_.
- [x] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### What went wrong, what did you observe?

Hello,

I recently found a bug where for a private poll with public link sharing enabled, where access by the owner account becomes broken during access to the public share link by the owner account. Here are the reproduction steps.
- I accessed the public share link I had enabled for the private poll, on a browser session where I was already logged in
- The first time I did this, nothing noticeable happened to my knowledge although there is a chance that it prevented the poll functioning for external/guest users as I was informed the poll "wasn't working". Under the hood, a new entry was added to the `oc_polls_share` database table.
- When I accessed the poll a second time from an already-logged in browser session via the public share link, another entry is added to the `oc_polls_share` table, and at this point the poll becomes inaccessible to the owner account. Notably two duplicate polls are shown in the account polls list, and the polls are listed as being 56 years old and private without any details available when I click on them through the user interface.
- When I manually deleted the database entries, the poll once again became accessible as normal through the nextcloud interface.
- I was able to consistently reproduce the bug by accessing the poll from the shared public link while logged into the owner account in the same browser session, every time.

Please see attached screenshot of the `oc_polls_share` database table after accessing the public share link in a browser session logged in by the poll owner account (note: name and email obscured for the user account)

<img width="1290" height="75" alt="Image" src="https://github.com/user-attachments/assets/33ce5aa0-f4bb-4792-8a25-5fa775082a33" />

See below screenshots for what the owner account sees when attempting to access the polls in the broken state:

<img width="1907" height="416" alt="Image" src="https://github.com/user-attachments/assets/9e9fcdf8-a8f4-4494-ba12-cfc0e0a2cd8c" />

<img width="1906" height="453" alt="Image" src="https://github.com/user-attachments/assets/4cc183be-b92a-4b32-a001-b123434ec043" />



### What did you expect, how polls should behave instead?

I expect the poll to remain accessible to the owner account regardless of how many times it accesses the poll by the public share link.

### What steps does it need to replay this bug?

1. Create a private poll with the following settings:

<img width="332" height="631" alt="Image" src="https://github.com/user-attachments/assets/d7b97613-bbca-405a-9930-d38cc83864fe" />

2. Add a public link under the sharing options
3. Create 2 votes by 2 unregistered/external guest users (note: probably optional but I have not tested otherwise)
4. Copy the public sharing link to clipboard and paste it in a new browser tab
5. Repeat step 4 once
6. You will notice that the poll appears broken, with a duplicate poll listed under "My Polls" and neither of these are accessible.


### Affected polls version

8.1.4

### Installation method

Installed/updated from the appstore (Apps section of your site)

### Installation type

First time installation

### Can you rule out that any extension you use is involved in the issue?

- [x] I have checked all browser extension

### Which browser did you use, when experiencing the bug?

- [x] Firefox
- [ ] Chrome
- [x] Chromium/Chromium based (i.e. Edge)
- [ ] Safari
- [ ] Other/Don't know

### Other browser

_No response_

### Add your browser log here

```json
NotificationsApp.vue:434 Notifications permissions not yet requested
main.ts:20   GET https://cloud.usepod.net/index.php/apps/polls/poll/1?time=1753171936575 500 (Internal Server Error)
(anonymous) @ adapters.js:67
xhr @ adapters.js:67
pd @ Axios.js:103
Promise.then
_request @ translation-DUYoTdjY.mjs:34
request @ Axios.js:229
(anonymous) @ AxiosHeaders.js:147
getFullPoll @ NcDashboardWidget-DJtdcW-W.mjs:181
load @ NcDashboardWidget-DJtdcW-W.mjs:181
R @ index.mjs:32
(anonymous) @ main.ts:20
(anonymous) @ index.mjs:32
Ir @ index.mjs:32
fu @ index.mjs:32
Ba.t.__weh.t.__weh @ index.mjs:32
fh @ index.mjs:32
gh @ index.mjs:32
Promise.then
ph @ index.mjs:32
mh @ index.mjs:32
o1 @ index.mjs:32
Fa.a.scheduler @ index.mjs:32
l.scheduler @ index.mjs:32
trigger @ index.mjs:32
k0 @ index.mjs:32
notify @ index.mjs:32
trigger @ index.mjs:32
set value @ index.mjs:32
x @ index.mjs:32
(anonymous) @ index.mjs:32
Promise.then
$ @ index.mjs:32
(anonymous) @ index.mjs:32
Promise.then
$ @ index.mjs:32
N @ index.mjs:32
install @ index.mjs:32
use @ index.mjs:32
(anonymous) @ main.ts:20
main.ts:20  [ERROR] polls: Error loading poll {app: 'polls', uid: 'trinity', level: 2, error: ke}app: "polls"error: kecode: "ERR_BAD_RESPONSE"config: {transitional: {…}, adapter: Array(3), transformRequest: Array(1), transformResponse: Array(1), timeout: 0, …}message: "Request failed with status code 500"name: "AxiosError"request: XMLHttpRequest {onreadystatechange: null, readyState: 4, timeout: 0, withCredentials: false, upload: XMLHttpRequestUpload, …}response: {data: '<!DOCTYPE html>\n<html class="ng-csp" data-placehol…a> – dive in\t\t\t</p>\n\t\t</footer>\n\t</body>\n</html>\n', status: 500, statusText: '', headers: Gt, config: {…}, …}status: 500stack: "AxiosError: Request failed with status code 500\n    at lm (https://cloud.usepod.net/apps/polls/js/NcAvatar-BadoF6cb-CdbDfRVm.chunk.mjs:8:1083)\n    at XMLHttpRequest.A (https://cloud.usepod.net/apps/polls/js/NcAvatar-BadoF6cb-CdbDfRVm.chunk.mjs:8:5718)\n    at Un.request (https://cloud.usepod.net/apps/polls/js/NcAvatar-BadoF6cb-CdbDfRVm.chunk.mjs:10:2069)\n    at async Proxy.load (https://cloud.usepod.net/apps/polls/js/NcDashboardWidget-DJtdcW-W-afeYk4Lg.chunk.mjs:1054:12446)"[[Prototype]]: Errorlevel: 2uid: "trinity"[[Prototype]]: Object
log @ AxiosHeaders.js:23
error @ AxiosHeaders.js:56
load @ NcDashboardWidget-DJtdcW-W.mjs:181
await in load
R @ index.mjs:32
(anonymous) @ main.ts:20
(anonymous) @ index.mjs:32
Ir @ index.mjs:32
fu @ index.mjs:32
Ba.t.__weh.t.__weh @ index.mjs:32
fh @ index.mjs:32
gh @ index.mjs:32
Promise.then
ph @ index.mjs:32
mh @ index.mjs:32
o1 @ index.mjs:32
Fa.a.scheduler @ index.mjs:32
l.scheduler @ index.mjs:32
trigger @ index.mjs:32
k0 @ index.mjs:32
notify @ index.mjs:32
trigger @ index.mjs:32
set value @ index.mjs:32
x @ index.mjs:32
(anonymous) @ index.mjs:32
Promise.then
$ @ index.mjs:32
(anonymous) @ index.mjs:32
Promise.then
$ @ index.mjs:32
N @ index.mjs:32
install @ index.mjs:32
use @ index.mjs:32
(anonymous) @ main.ts:20
adapters.js:67  Uncaught (in promise) ke {message: 'Request failed with status code 500', name: 'AxiosError', code: 'ERR_BAD_RESPONSE', config: {…}, request: XMLHttpRequest, …}
lm @ adapters.js:67
A @ adapters.js:67
XMLHttpRequest.send
(anonymous) @ adapters.js:67
xhr @ adapters.js:67
pd @ Axios.js:103
Promise.then
_request @ translation-DUYoTdjY.mjs:34
request @ Axios.js:229
(anonymous) @ AxiosHeaders.js:147
getFullPoll @ NcDashboardWidget-DJtdcW-W.mjs:181
load @ NcDashboardWidget-DJtdcW-W.mjs:181
R @ index.mjs:32
(anonymous) @ main.ts:20
(anonymous) @ index.mjs:32
Ir @ index.mjs:32
fu @ index.mjs:32
Ba.t.__weh.t.__weh @ index.mjs:32
fh @ index.mjs:32
gh @ index.mjs:32
Promise.then
ph @ index.mjs:32
mh @ index.mjs:32
o1 @ index.mjs:32
Fa.a.scheduler @ index.mjs:32
l.scheduler @ index.mjs:32
trigger @ index.mjs:32
k0 @ index.mjs:32
notify @ index.mjs:32
trigger @ index.mjs:32
set value @ index.mjs:32
x @ index.mjs:32
(anonymous) @ index.mjs:32
Promise.then
$ @ index.mjs:32
(anonymous) @ index.mjs:32
Promise.then
$ @ index.mjs:32
N @ index.mjs:32
install @ index.mjs:32
use @ index.mjs:32
(anonymous) @ main.ts:20
```

### Additional client environment information

Initially tested in Firefox with some extensions; tested again and achieved the same results in Microsoft Edge without any extensions/plugins installed.

### NC version

Nextcloud 31

### Other Nextcloud version

_No response_

### PHP engine version

PHP 8.3

### Other PHP version

_No response_

### Database engine

MySQL

### Database Engine version or other Database

_No response_

### Which user-backends are you using?

- [x] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other/Don't know

### Add your nextcloud server log here

```json
{"reqId":"T4TVsl1gzhBbNF7wmokt","level":3,"time":"2025-07-22T08:12:17+00:00","remoteAddr":"2a00:23c5:f1a6:7e01:f9a4:5041:8bbf:7531","user":"trinity","app":"index","method":"GET","url":"/index.php/apps/polls/poll/1?time=1753171936575","message":"Did not expect more than one result when executing: query \"SELECT `polls_polls`.*, MAX(options.timestamp) AS max_date, coalesce(user_shares.type, '') AS user_role, `user_shares`.`locked` AS `is_current_user_locked`, coalesce(user_shares.token, '') AS share_token, group_concat(distinct group_shares.user_id SEPARATOR \",\") AS group_shares, group_concat(distinct poll_groups.group_id SEPARATOR \",\") AS poll_groups, group_concat(distinct poll_group_shares.type SEPARATOR \",\") AS poll_group_user_shares, COUNT(DISTINCT(votes.id)) AS current_user_votes, COUNT(DISTINCT(CASE WHEN votes.vote_answer = 'yes' THEN votes.id END)) AS current_user_votes_yes, COUNT(DISTINCT(CASE WHEN votes.vote_answer = 'no' THEN votes.id END)) AS current_user_votes_no, COUNT(DISTINCT(CASE WHEN votes.vote_answer = 'maybe' THEN votes.id END)) AS current_user_votes_maybe, COUNT(DISTINCT(CASE WHEN vote_options_sub.id is NULL THEN votes.id END)) AS current_user_orphaned_votes, COUNT(DISTINCT(participants.user_id)) AS participants_count FROM `*PREFIX*polls_polls` `polls_polls` LEFT JOIN `*PREFIX*polls_options` `options` ON (`options`.`poll_id` = `polls_polls`.`id`) AND (`options`.`deleted` = '0') LEFT JOIN `*PREFIX*polls_share` `user_shares` ON (`user_shares`.`poll_id` = `polls_polls`.`id`) AND (`user_shares`.`user_id` = :dcValue1) AND (`user_shares`.`deleted` = '0') LEFT JOIN `*PREFIX*polls_share` `group_shares` ON (`group_shares`.`poll_id` = `polls_polls`.`id`) AND (`group_shares`.`type` = 'group') AND (`group_shares`.`deleted` = '0') LEFT JOIN `*PREFIX*polls_groups_polls` `poll_groups` ON `polls_polls`.`id` = `poll_groups`.`poll_id` LEFT JOIN `*PREFIX*polls_votes` `votes` ON (`votes`.`poll_id` = `polls_polls`.`id`) AND (`votes`.`user_id` = :dcValue3) LEFT JOIN `*PREFIX*polls_votes` `participants` ON `participants`.`poll_id` = `polls_polls`.`id` LEFT JOIN `*PREFIX*polls_share` `poll_group_shares` ON (`poll_group_shares`.`group_id` = `poll_groups`.`group_id`) AND (`poll_group_shares`.`deleted` = '0') AND (`poll_group_shares`.`user_id` = :dcValue2) LEFT JOIN `*PREFIX*polls_options` `vote_options_sub` ON (`vote_options_sub`.`poll_id` = `votes`.`poll_id`) AND (`vote_options_sub`.`poll_option_text` = `votes`.`vote_option_text`) AND (`vote_options_sub`.`deleted` = '0') WHERE `polls_polls`.`id` = :dcValue4 GROUP BY `polls_polls`.`id`, `user_shares`.`type`, `user_shares`.`locked`, `user_shares`.`token`\"; ","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36 Edg/138.0.0.0","version":"31.0.7.1","exception":{"Exception":"OCP\\AppFramework\\Db\\MultipleObjectsReturnedException","Message":"Did not expect more than one result when executing: query \"SELECT `polls_polls`.*, MAX(options.timestamp) AS max_date, coalesce(user_shares.type, '') AS user_role, `user_shares`.`locked` AS `is_current_user_locked`, coalesce(user_shares.token, '') AS share_token, group_concat(distinct group_shares.user_id SEPARATOR \",\") AS group_shares, group_concat(distinct poll_groups.group_id SEPARATOR \",\") AS poll_groups, group_concat(distinct poll_group_shares.type SEPARATOR \",\") AS poll_group_user_shares, COUNT(DISTINCT(votes.id)) AS current_user_votes, COUNT(DISTINCT(CASE WHEN votes.vote_answer = 'yes' THEN votes.id END)) AS current_user_votes_yes, COUNT(DISTINCT(CASE WHEN votes.vote_answer = 'no' THEN votes.id END)) AS current_user_votes_no, COUNT(DISTINCT(CASE WHEN votes.vote_answer = 'maybe' THEN votes.id END)) AS current_user_votes_maybe, COUNT(DISTINCT(CASE WHEN vote_options_sub.id is NULL THEN votes.id END)) AS current_user_orphaned_votes, COUNT(DISTINCT(participants.user_id)) AS participants_count FROM `*PREFIX*polls_polls` `polls_polls` LEFT JOIN `*PREFIX*polls_options` `options` ON (`options`.`poll_id` = `polls_polls`.`id`) AND (`options`.`deleted` = '0') LEFT JOIN `*PREFIX*polls_share` `user_shares` ON (`user_shares`.`poll_id` = `polls_polls`.`id`) AND (`user_shares`.`user_id` = :dcValue1) AND (`user_shares`.`deleted` = '0') LEFT JOIN `*PREFIX*polls_share` `group_shares` ON (`group_shares`.`poll_id` = `polls_polls`.`id`) AND (`group_shares`.`type` = 'group') AND (`group_shares`.`deleted` = '0') LEFT JOIN `*PREFIX*polls_groups_polls` `poll_groups` ON `polls_polls`.`id` = `poll_groups`.`poll_id` LEFT JOIN `*PREFIX*polls_votes` `votes` ON (`votes`.`poll_id` = `polls_polls`.`id`) AND (`votes`.`user_id` = :dcValue3) LEFT JOIN `*PREFIX*polls_votes` `participants` ON `participants`.`poll_id` = `polls_polls`.`id` LEFT JOIN `*PREFIX*polls_share` `poll_group_shares` ON (`poll_group_shares`.`group_id` = `poll_groups`.`group_id`) AND (`poll_group_shares`.`deleted` = '0') AND (`poll_group_shares`.`user_id` = :dcValue2) LEFT JOIN `*PREFIX*polls_options` `vote_options_sub` ON (`vote_options_sub`.`poll_id` = `votes`.`poll_id`) AND (`vote_options_sub`.`poll_option_text` = `votes`.`vote_option_text`) AND (`vote_options_sub`.`deleted` = '0') WHERE `polls_polls`.`id` = :dcValue4 GROUP BY `polls_polls`.`id`, `user_shares`.`type`, `user_shares`.`locked`, `user_shares`.`token`\"; ","Code":0,"Trace":[{"file":"/var/www/ud23_157/html/cloud/lib/public/AppFramework/Db/QBMapper.php","line":375,"function":"findOneQuery","class":"OCP\\AppFramework\\Db\\QBMapper","type":"->"},{"file":"/var/www/ud23_157/html/cloud/apps/polls/lib/Db/PollMapper.php","line":73,"function":"findEntity","class":"OCP\\AppFramework\\Db\\QBMapper","type":"->"},{"file":"/var/www/ud23_157/html/cloud/apps/polls/lib/Service/PollService.php","line":176,"function":"find","class":"OCA\\Polls\\Db\\PollMapper","type":"->"},{"file":"/var/www/ud23_157/html/cloud/apps/polls/lib/Controller/PollController.php","line":111,"function":"get","class":"OCA\\Polls\\Service\\PollService","type":"->"},{"file":"/var/www/ud23_157/html/cloud/apps/polls/lib/Controller/PollController.php","line":105,"function":"getFullPoll","class":"OCA\\Polls\\Controller\\PollController","type":"->"},{"file":"/var/www/ud23_157/html/cloud/apps/polls/lib/Controller/BaseController.php","line":42,"function":"OCA\\Polls\\Controller\\{closure}","class":"OCA\\Polls\\Controller\\PollController","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/ud23_157/html/cloud/apps/polls/lib/Controller/PollController.php","line":105,"function":"response","class":"OCA\\Polls\\Controller\\BaseController","type":"->"},{"file":"/var/www/ud23_157/html/cloud/lib/private/AppFramework/Http/Dispatcher.php","line":200,"function":"getFull","class":"OCA\\Polls\\Controller\\PollController","type":"->"},{"file":"/var/www/ud23_157/html/cloud/lib/private/AppFramework/Http/Dispatcher.php","line":114,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/ud23_157/html/cloud/lib/private/AppFramework/App.php","line":161,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/ud23_157/html/cloud/lib/private/Route/Router.php","line":315,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/ud23_157/html/cloud/lib/base.php","line":1040,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/ud23_157/html/cloud/index.php","line":24,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/ud23_157/html/cloud/lib/public/AppFramework/Db/QBMapper.php","Line":286,"message":"Did not expect more than one result when executing: query \"SELECT `polls_polls`.*, MAX(options.timestamp) AS max_date, coalesce(user_shares.type, '') AS user_role, `user_shares`.`locked` AS `is_current_user_locked`, coalesce(user_shares.token, '') AS share_token, group_concat(distinct group_shares.user_id SEPARATOR \",\") AS group_shares, group_concat(distinct poll_groups.group_id SEPARATOR \",\") AS poll_groups, group_concat(distinct poll_group_shares.type SEPARATOR \",\") AS poll_group_user_shares, COUNT(DISTINCT(votes.id)) AS current_user_votes, COUNT(DISTINCT(CASE WHEN votes.vote_answer = 'yes' THEN votes.id END)) AS current_user_votes_yes, COUNT(DISTINCT(CASE WHEN votes.vote_answer = 'no' THEN votes.id END)) AS current_user_votes_no, COUNT(DISTINCT(CASE WHEN votes.vote_answer = 'maybe' THEN votes.id END)) AS current_user_votes_maybe, COUNT(DISTINCT(CASE WHEN vote_options_sub.id is NULL THEN votes.id END)) AS current_user_orphaned_votes, COUNT(DISTINCT(participants.user_id)) AS participants_count FROM `*PREFIX*polls_polls` `polls_polls` LEFT JOIN `*PREFIX*polls_options` `options` ON (`options`.`poll_id` = `polls_polls`.`id`) AND (`options`.`deleted` = '0') LEFT JOIN `*PREFIX*polls_share` `user_shares` ON (`user_shares`.`poll_id` = `polls_polls`.`id`) AND (`user_shares`.`user_id` = :dcValue1) AND (`user_shares`.`deleted` = '0') LEFT JOIN `*PREFIX*polls_share` `group_shares` ON (`group_shares`.`poll_id` = `polls_polls`.`id`) AND (`group_shares`.`type` = 'group') AND (`group_shares`.`deleted` = '0') LEFT JOIN `*PREFIX*polls_groups_polls` `poll_groups` ON `polls_polls`.`id` = `poll_groups`.`poll_id` LEFT JOIN `*PREFIX*polls_votes` `votes` ON (`votes`.`poll_id` = `polls_polls`.`id`) AND (`votes`.`user_id` = :dcValue3) LEFT JOIN `*PREFIX*polls_votes` `participants` ON `participants`.`poll_id` = `polls_polls`.`id` LEFT JOIN `*PREFIX*polls_share` `poll_group_shares` ON (`poll_group_shares`.`group_id` = `poll_groups`.`group_id`) AND (`poll_group_shares`.`deleted` = '0') AND (`poll_group_shares`.`user_id` = :dcValue2) LEFT JOIN `*PREFIX*polls_options` `vote_options_sub` ON (`vote_options_sub`.`poll_id` = `votes`.`poll_id`) AND (`vote_options_sub`.`poll_option_text` = `votes`.`vote_option_text`) AND (`vote_options_sub`.`deleted` = '0') WHERE `polls_polls`.`id` = :dcValue4 GROUP BY `polls_polls`.`id`, `user_shares`.`type`, `user_shares`.`locked`, `user_shares`.`token`\"; ","exception":[],"CustomMessage":"Did not expect more than one result when executing: query \"SELECT `polls_polls`.*, MAX(options.timestamp) AS max_date, coalesce(user_shares.type, '') AS user_role, `user_shares`.`locked` AS `is_current_user_locked`, coalesce(user_shares.token, '') AS share_token, group_concat(distinct group_shares.user_id SEPARATOR \",\") AS group_shares, group_concat(distinct poll_groups.group_id SEPARATOR \",\") AS poll_groups, group_concat(distinct poll_group_shares.type SEPARATOR \",\") AS poll_group_user_shares, COUNT(DISTINCT(votes.id)) AS current_user_votes, COUNT(DISTINCT(CASE WHEN votes.vote_answer = 'yes' THEN votes.id END)) AS current_user_votes_yes, COUNT(DISTINCT(CASE WHEN votes.vote_answer = 'no' THEN votes.id END)) AS current_user_votes_no, COUNT(DISTINCT(CASE WHEN votes.vote_answer = 'maybe' THEN votes.id END)) AS current_user_votes_maybe, COUNT(DISTINCT(CASE WHEN vote_options_sub.id is NULL THEN votes.id END)) AS current_user_orphaned_votes, COUNT(DISTINCT(participants.user_id)) AS participants_count FROM `*PREFIX*polls_polls` `polls_polls` LEFT JOIN `*PREFIX*polls_options` `options` ON (`options`.`poll_id` = `polls_polls`.`id`) AND (`options`.`deleted` = '0') LEFT JOIN `*PREFIX*polls_share` `user_shares` ON (`user_shares`.`poll_id` = `polls_polls`.`id`) AND (`user_shares`.`user_id` = :dcValue1) AND (`user_shares`.`deleted` = '0') LEFT JOIN `*PREFIX*polls_share` `group_shares` ON (`group_shares`.`poll_id` = `polls_polls`.`id`) AND (`group_shares`.`type` = 'group') AND (`group_shares`.`deleted` = '0') LEFT JOIN `*PREFIX*polls_groups_polls` `poll_groups` ON `polls_polls`.`id` = `poll_groups`.`poll_id` LEFT JOIN `*PREFIX*polls_votes` `votes` ON (`votes`.`poll_id` = `polls_polls`.`id`) AND (`votes`.`user_id` = :dcValue3) LEFT JOIN `*PREFIX*polls_votes` `participants` ON `participants`.`poll_id` = `polls_polls`.`id` LEFT JOIN `*PREFIX*polls_share` `poll_group_shares` ON (`poll_group_shares`.`group_id` = `poll_groups`.`group_id`) AND (`poll_group_shares`.`deleted` = '0') AND (`poll_group_shares`.`user_id` = :dcValue2) LEFT JOIN `*PREFIX*polls_options` `vote_options_sub` ON (`vote_options_sub`.`poll_id` = `votes`.`poll_id`) AND (`vote_options_sub`.`poll_option_text` = `votes`.`vote_option_text`) AND (`vote_options_sub`.`deleted` = '0') WHERE `polls_polls`.`id` = :dcValue4 GROUP BY `polls_polls`.`id`, `user_shares`.`type`, `user_shares`.`locked`, `user_shares`.`token`\"; "},"id":"687f4965172c1"}
```

### Additional environment informations

_No response_

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.usepod.net"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "31.0.7.1",
        "overwrite.cli.url": "https:\/\/cloud.usepod.net",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "maintenance_window_start": 1,
        "default_phone_region": "GB",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpsecure": "ssl",
        "mail_smtpauth": true,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "versions_retention_obligation": "auto, 30",
        "app_install_overwrite": []
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 4.0.0
  - admin_audit: 1.21.0
  - app_api: 5.0.2
  - bruteforcesettings: 4.0.0
  - calendar: 5.3.6
  - circles: 31.0.0
  - cloud_federation_api: 1.14.0
  - collectives: 3.0.2
  - comments: 1.21.0
  - contacts: 7.2.0
  - contactsinteraction: 1.12.0
  - cookbook: 0.11.3
  - dashboard: 7.11.0
  - dav: 1.33.0
  - federatedfilesharing: 1.21.0
  - federation: 1.21.0
  - files: 2.3.1
  - files_downloadlimit: 4.0.0
  - files_pdfviewer: 4.0.0
  - files_reminders: 1.4.0
  - files_sharing: 1.23.1
  - files_trashbin: 1.21.0
  - files_versions: 1.24.0
  - firstrunwizard: 4.0.0
  - forms: 5.1.2
  - gestion: 2.8.6
  - logreader: 4.0.0
  - lookup_server_connector: 1.19.0
  - mail: 5.1.8
  - news: 26.0.2
  - nextcloud_announcements: 3.0.0
  - notes: 4.12.2
  - notifications: 4.0.0
  - oauth2: 1.19.1
  - password_policy: 3.0.0
  - photos: 4.0.0
  - polls: 8.1.4
  - privacy: 3.0.0
  - profile: 1.0.0
  - provisioning_api: 1.21.0
  - recommendations: 4.0.0
  - related_resources: 2.0.0
  - serverinfo: 3.0.0
  - settings: 1.14.0
  - sharebymail: 1.21.0
  - spreed: 21.1.2
  - support: 3.0.0
  - survey_client: 3.0.0
  - systemtags: 1.21.1
  - text: 5.0.0
  - theming: 2.6.1
  - twofactor_backupcodes: 1.20.0
  - updatenotification: 1.21.0
  - user_status: 1.11.0
  - viewer: 4.0.0
  - weather_status: 1.11.0
  - webhook_listeners: 1.2.0
  - workflowengine: 2.13.0
Disabled:
  - encryption: 2.19.0
  - files_external: 1.23.0
  - suspicious_login: 9.0.1
  - twofactor_nextcloud_notification: 5.0.0
  - twofactor_totp: 13.0.0-dev.0
  - user_ldap: 1.22.0
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Additional Information

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -22,6 +22,7 @@ All notable changes to this project will be documented in this file.
  - Bring back indicator for confirmed options after closing the poll
  - Add CSP to allow worker
  - fix loading archived polls
+ - broken access if a logged in user by public share enters a poll he already had access to
 
 ## [8.1.4] - 2025-07-15
 ### Fixes

--- a/lib/Service/ShareService.php
+++ b/lib/Service/ShareService.php
@@ -161,6 +161,19 @@ class ShareService {
 			$this->share->setDisplayName('');
 		}
 
+		$poll = $this->pollMapper->get($this->share->getPollId(), withRoles: true);
+
+		if ($poll->getIsAllowed(Poll::PERMISSION_VOTE_EDIT)
+			|| $poll->getIsAllowed(Poll::PERMISSION_POLL_EDIT)
+		) {
+			// user is allowed to access the poll, continue without creating a new share
+			return $this->share;
+		}
+
+		if (!$poll->getIsAllowed(Poll::PERMISSION_POLL_VIEW)) {
+			throw new ForbiddenException('User is not allowed to access this poll');
+		}
+
 		// Exception: logged in user, accesses the poll via public share link
 		if ($this->share->getType() === Share::TYPE_PUBLIC && $this->userSession->getIsLoggedIn()) {
 			$this->convertPublicShareToPersonalShare();

